packages/runtime/src/state/modules/api-resources.ts

```diff
@@ -134,7 +134,9 @@ export function getAPIResource<T extends APIResourceType>(
     : null
 }
 
-export function reducer(state: State = getInitialState(), action: Action): State {
+export function reducer(state: State = getInitialState(), action: Action | UnknownAction): State {
+  if (!isKnownAction(action)) return state
+
```

Comment:
Here and below, adjust reducer's type to acknowledge that it might receive an unknown action, pre-filter actions using the `isKnownAction` predicate to keep the `switch`/`case` below properly typed.

In contrast to the middleware, the `isKnownAction` check has to happen *inside* of the reducer to allow for the state slice initialization (the `state: State = getInitialState()` default in the reducer definition).